taskcluster/xpi_taskgraph/worker_types.py

```diff
@@ -100,3 +102,78 @@ def build_github_release_payload(config, task, task_def):
             "{}:github:action:{}".format(scope_prefix, worker["action"]),
         ]
     )
+
+
+@payload_builder(
+    "scriptworker-beetmover",
+    schema={
+        Required("action-scope"): str,
+        Required("bucket-scope"): str,
+        Required("artifact-map"): [
+            {
+                Required("paths"): {
+                    Any(str): {
+                        Required("destinations"): [str],
+                    },
+                },
+                Required("taskId"): taskref_or_string,
+            }
+        ],
+        Required("release-properties"): {
+            Required("app-name"): str,
+            Required("app-version"): str,
+            Required("branch"): str,
+            Required("build-id"): str,
+            Optional("hash-type"): str,
+            Optional("platform"): str,
+        },
+        Required("upstream-artifacts"): [
+            {
+                Required("locale"): str,
+                Required("taskId"): taskref_or_string,
+                Required("taskType"): str,
+                Required("paths"): [str],
+            }
+        ],
+    },
+)
+def build_scriptworker_beetmover_payload(config, task, task_def):
+    worker = task["worker"]
+    task_def["tags"]["worker-implementation"] = "scriptworker"
+    artifact_map = worker["artifact-map"]
+    for map_ in artifact_map:
+        map_["locale"] = "multi"
+        for path_config in map_["paths"].values():
+            path_config["checksums_path"] = ""
```

Comment:
We may want to populate this, not sure.

Essentially, in this [gecko beetmover task](https://firefox-ci-tc.services.mozilla.com/tasks/S2hsSOyCTxGAxPFhhXxnVg#artifacts), we upload artifacts called `target.checksums` and `manifest.json`. These contain checksums (e.g. sha256, sha512) and filesize, which we'll need in the balrog task.

I'm a bit fuzzy as to what specific incantation we need to generate those, and how the balrog task consumes them. I'm happy to leave this piece for another PR, but I suspect it'll block the balrog PR.

Reply:
They are generated in morgoth. It seems straight forward. But in morgoth's case the file is in memory, so it can read it and generate that information. How can we get those file in memory to generate the SHA? Do we generate at the graph generation stage?